coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala

```diff
@@ -1,131 +1,133 @@
     val execPlans = planners.values.toList.distinct.map(_.materialize(logicalPlan, qContext))
-    if (execPlans.size == 1) execPlans.head
-    else LabelValuesDistConcatExec(qContext, inProcessPlanDispatcher, execPlans)
+    PlanResult(Seq(if (execPlans.size == 1) execPlans.head
+    else LabelValuesDistConcatExec(qContext, inProcessPlanDispatcher, execPlans)))
   }
 
-  private def materializeLabelNames(logicalPlan: LogicalPlan, qContext: QueryContext) = {
+  private def materializeLabelNames(logicalPlan: LogicalPlan, qContext: QueryContext): PlanResult = {
     val execPlans = planners.values.toList.distinct.map(_.materialize(logicalPlan, qContext))
-    if (execPlans.size == 1) execPlans.head
-    else LabelNamesDistConcatExec(qContext, inProcessPlanDispatcher, execPlans)
+    PlanResult(Seq(if (execPlans.size == 1) execPlans.head
+    else LabelNamesDistConcatExec(qContext, inProcessPlanDispatcher, execPlans)))
   }
 
-  private def materializeSeriesKeysFilters(logicalPlan: LogicalPlan, qContext: QueryContext) = {
+  private def materializeSeriesKeysFilters(logicalPlan: LogicalPlan, qContext: QueryContext): PlanResult = {
     val execPlans = planners.values.toList.distinct.map(_.materialize(logicalPlan, qContext))
-    if (execPlans.size == 1) execPlans.head
-    else PartKeysDistConcatExec(qContext, inProcessPlanDispatcher, execPlans)
+    PlanResult(Seq(if (execPlans.size == 1) execPlans.head
+    else PartKeysDistConcatExec(qContext, inProcessPlanDispatcher, execPlans)))
   }
 
-  private def materializeTsCardinalities(logicalPlan: TsCardinalities, qContext: QueryContext): ExecPlan = {
-    // TODO: this is a hacky fix to prevent delegation to planners with reduce-incompatible data.
-    planners.find{case (name, _) => name == defaultPlanner}.get._2.materialize(logicalPlan, qContext)
+  private def materializeTsCardinalities(logicalPlan: TsCardinalities, qContext: QueryContext): PlanResult = {
+    // Delegate to defaultPlanner
+    planners.get(defaultPlanner).map(p => PlanResult(Seq(p.materialize(logicalPlan, qContext))))
+    .getOrElse(PlanResult(Seq()))
   }
+
+
+  /**
+   * Apart from some metadata queries, SinglePartitionPlanner should either push down to underlying planner or perform
+   * the operation inprocess if these operations span across different planners. The implementation execute plans
+   * inprocess are defined in DefaultPlanner and thus we delegate the materialization of these plan to the default
+   * walk implementation in DefaultPlanner
+   *
+   * @param logicalPlan    The LogicalPlan instance
+   * @param qContext       The QueryContext
+   * @param forceInProcess if true, all materialized plans for this entire
+   *                       logical plan will dispatch via an InProcessDispatcher
+   * @return The PlanResult containing the ExecPlan
+   */
+  override def walkLogicalPlanTree(logicalPlan: LogicalPlan, qContext: QueryContext, forceInProcess: Boolean)
```

Comment:
A+ for this method alone-- thanks for cleaning everything up and getting rid of that copy-paste code from the `DefaultPlanner`.